vault/diagnose/tls_verification.go

```diff
@@ -102,23 +166,45 @@ func TLSFileChecks(certFilePath, keyFilePath string) error {
 
 	// Verify checks that certificate isn't expired, is of correct usage type, and has an appropriate
 	// chain.
-	_, err = leafCerts[0].Verify(x509.VerifyOptions{
+	_, err := leafCerts[0].Verify(x509.VerifyOptions{
 		Roots:         rootPool,
 		Intermediates: interPool,
 		KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
 	})
 
 	if err != nil {
-		return fmt.Errorf("failed to verify certificate: %w", err)
+		return fmt.Errorf("failed to verify primary provided leaf certificate: %w", err)
 	}
 
-	// After verify passes, we need to check the values on the certificate itself.
-	// This is a separate check beyond the certificate expiry and chain checks.
+	return nil
+}
 
-	_, err = tls.LoadX509KeyPair(certFilePath, keyFilePath)
-	if err != nil {
-		return err
+func TLSFileWarningChecks(leafcerts, interCerts, rootCerts []*x509.Certificate) ([]string, error) {
+	var warnings []string
+	for _, c := range leafcerts {
+		if NearExpiration(c) {
+			warnings = append(warnings, fmt.Sprintf("certificate %d is expired or near expiry", c.SerialNumber))
```

Comment:
Since the messages in this section are all similar, would it make sense to add cert type in the message as well? Like here, we can add "Leaf certificate ... ". This would make it easier to spot the issue.

Reply:
Makes sense to me!